rst_files/markov_asset.rst

```diff
@@ -589,57 +543,40 @@ the type `AssetPriceModel`
     n = 25
     default_mc = tauchen(n, ρ, σ)
 
-    mutable struct AssetPriceModel{TF<:AbstractFloat, TI<:Integer}
-        β::TF            # Discount factor
-        γ::TF            # Coefficient of risk aversion
-        mc::MarkovChain  # State process
-        n::TI            # Number of states
-        g::Function      # Function mapping states into growth rates
-    end
-
-    function AssetPriceModel(;β::AbstractFloat = 0.96,
-                              γ::AbstractFloat = 2.0,
-                              mc::MarkovChain = default_mc,
-                              g::Function = exp)
-        n = size(mc.p)[1]
-        return AssetPriceModel(β, γ, mc, n, g)
-    end
-
+    AssetPriceModel = @with_kw (β = 0.96,
+                                γ = 2.0,
+                                mc = default_mc,
+                                n = size(mc.p)[1],
+                                g = exp)
 
-    # test stability of matrix Q 
-    function test_stability(ap::AssetPriceModel, Q::Matrix)
+    # test stability of matrix Q
+    function test_stability(ap, Q)
         sr = maximum(abs, eigvals(Q))
         if sr ≥ 1 / ap.β
             msg = "Spectral radius condition failed with radius = $sr"
             throw(ArgumentError(msg))
         end
     end
 
-
-    # price/dividend ratio of the Lucas tree 
-    function tree_price(ap::AssetPriceModel)
-        # == Simplify names, set up matrices  == #
-        β, γ, P, y = ap.β, ap.γ, ap.mc.p, ap.mc.state_values
+    # price/dividend ratio of the Lucas tree
+    function tree_price(ap; γ = ap.γ)
+        # Simplify names, set up matrices
+        β, P, y = ap.β, ap.mc.p, ap.mc.state_values
```

Comment:
This should be an `@unpack`, i.e. 

```
@unpack β, P, state_values = ap 
```

Reply:
That struct uses `ap.mc.val`, not `ap.val` so `@unpack` doesn't work in this case.

Reply:
I think we should unpack to beta and mc, and then can do the mc stuff in a separate line. I think it would be more clear and consistent with your style